lib/posthog/feature_flags.ex

```diff
@@ -0,0 +1,239 @@
+            {:ok, true}
+
+          %{"flags" => %{^flag_name => _}} ->
+            {:ok, false}
+
+          %{"flags" => _} ->
+            {:error,
+             %PostHog.UnexpectedResponseError{
+               response: body,
+               message: "Feature flag #{flag_name} was not found in the response"
+             }}
+        end
+
+      # Make sure we keep track of the feature flag usage for debugging purposes
+      # Users are NOT charged extra for this, but it's still good to have.
+      log_feature_flag_usage(name, distinct_id, flag_name, result)
+
+      result
+    end
+  end
+
+  @doc """
+  Checks feature flag and returns the variant or raises on error.
+
+  This is a wrapper around `check/3` that returns the variant directly
+  or raises an exception on error. This follows the Elixir convention where
+  functions ending with `!` raise exceptions instead of returning error tuples.
+
+  > **Warning**: Use this function with care as it will raise an error if the feature flag
+  > is not found or if there are any API errors. For more resilient code, use `check/3`
+  > which returns `{:error, reason}` instead of raising.
+
+  ## Examples
+
+  Check feature flag and get the variant:
+
+      iex> PostHog.FeatureFlags.check!("example-feature-flag-1", "user123")
+      true
+
+  Check multivariant feature flag for distinct_id in current context:
+
+      iex> PostHog.set_context(%{distinct_id: "user123"})
+      iex> PostHog.FeatureFlags.check!("example-feature-flag-1")
+      "variant1"
+
+  Check feature flag through a named PostHog instance:
+
+      iex> PostHog.FeatureFlags.check!(MyPostHog, "example-feature-flag-1", "user123")
+      false
+
+  Raises an error when feature flag is not found:
+
+      iex> PostHog.FeatureFlags.check!("example-feature-flag-3", "user123")
+      ** (PostHog.UnexpectedResponseError) Feature flag example-feature-flag-3 was not found in the response
+  """
+  @spec check!(supervisor_name(), String.t(), distinct_id() | map() | nil) :: boolean() | String.t() | no_return()
+  def check!(name \\ PostHog, flag_name, distinct_id_or_body \\ nil) do
+    case check(name, flag_name, distinct_id_or_body) do
+      {:ok, result} -> result
+      {:error, error} -> raise error
+    end
+  end
+
+  @doc false
+  @spec log_feature_flag_usage(supervisor_name(), String.t(), String.t(), {:ok, boolean() | String.t()} | {:error, Exception.t()}) :: :ok
+  defp log_feature_flag_usage(name, distinct_id, flag_name, result)
+  defp log_feature_flag_usage(name, distinct_id, flag_name, {:ok, variant}) do
+    PostHog.capture(name, "$feature_flag_called", %{
+      distinct_id: distinct_id,
+      "$feature_flag": flag_name,
+      "$feature_flag_response": variant
+    })
+
+    PostHog.set_context(name, %{"$feature/#{flag_name}" => variant})
+  end
+
+  defp log_feature_flag_usage(_name, _distinct_id, _flag_name, {:error, _error}) do
+    # Do nothing for error cases
+    :ok
+  end
+
+  @doc false
```

Comment:
*sigh*
fun fact, if you _do_ write a `@doc` for a private function, compiler will issue a warning